### DISCUSSION TITLE
How to implement a "user rating" feature?

### DISCUSSION THREAD

**Discussion by vibl** [directus/directus#16637] [Archive]
How can I create something akin to a "user rating" feature? I.e., each user can create and edit their own M2M row (which includes a rating integer field) for each article.

I created a many-to-many relationship article_user_rating but I couldn't find a way to constrain the [article_id, user_id] pair to be unique (in the GUI, in the doc, here in the GH discussions or in Discord). I guess I could manually add a composite key in the database. It seems to be supported in Directus according to https://github.com/directus/directus/issues/15063. Can someone confirm?

Also, I couldn't manage to display the 5-stars widget for the current user in the article edit page.

Can someone help?

**Comment by cdwmhcc** [comment]
This function is relatively easy to implement, but I am more curious about how to calculate the average score if I have hundreds of thousands of data?
